src/main_functions_home.c

```diff
@@ -1,110 +1,151 @@
-#include <sys/stat.h>
+#include <stdlib.h>
 #include <stdio.h>
-#include <stdlib.h>	
+#include <stdbool.h>
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <string.h>
-#include <stdbool.h>
 #include <assert.h>
 #include "shellstrings.h"
 #include "parser.h"
 #include "main_functions_home.h"
+#include "log.c/src/log.h"
 
 /* See main_functions_home.h */
-void help_page()
-{
-    shell_help();
-	shell_prompt();
+void help_page(bool *color) {
+    shell_help(color);
+	shell_prompt(color);
 
-	parse_read_line();      // accept any input in the command line
+	// Accept any input in the command line
+	parse_read_line();
+	log_trace("exiting the help page.");
 }
 
 /* See main_functions_home.h */
-bool fileexists(const char* filename)
-{
+bool fileexists(const char *filename) {
 	struct stat buffer;
-	return (bool)(stat(filename, &buffer) == 0);
+
+	if (stat(filename, &buffer) == 0) {
+		return true;
+	}
+	return false;
 }
 
 /* See main_functions_home.h */
-int change_mode(char* arg)
-{
+int change_mode(char *arg, bool* color) {
+	log_trace("change_mode mode type: %s", arg);
 	int a = atoi(arg);
-    if ((a == 1) | (a == 2) | (a == 3)) {
+
+    if ((a == QUIET_MODE) || (a == VERBOSE_MODE) || (a == INTERACTIVE_MODE)) {
         return a;
-	} else {
-        shell_error("Input invalid: return to mode 3, interactive mode");		
-	    return 3; 		// default is 3, given this function is only called in main_page, at which point interactive is probably what user intended
-    }
-    return 3;
+	}
+
+    shell_error("change_mode mode type invalid; reverting to interactive mode.", color);
+    /*
+     * The default is 3, given this function is only called in main_page(),
+     * at which point interactive mode is what user probably intended
+     */
+	return INTERACTIVE_MODE;
 }
 
 /* See main_functions_home.h */
-void main_page(bool* quit, int *mode, char* file_name, char* dict_name)
-{
-	char* line;
-	char** args;
+void main_page(bool *quit, int *mode, char *filename, char *dict, bool *color) {
```

Comment:
Why is this a `bool *` and not just a `bool`?

Reply:
`*quit` and `*color` are booleans called here because the main menu can change their value, and if I just called the value, I wouldn't be able to change it to affect future prints to output. main_page is one of a few functions that can change that value.
I've changed the function calls for functions who won't affect the value, such as the many you commented on for shellstrings, but I think for main_page it might be necessary to do `*quit` and `*color`.